src/main/java/com/espro/flink/consul/metric/ConsulMetricService.java

```diff
@@ -0,0 +1,113 @@
+package com.espro.flink.consul.metric;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.Metric;
+import org.apache.flink.metrics.SimpleCounter;
+import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.DescriptiveStatisticsHistogram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+/**
+ * This class registers metrics to Flink metrics and then calculates and publishes them.
+ */
+public class ConsulMetricService{
```

Comment:
I still can't see the benefit of having a central metric service in comparison to handle metrics per service. This kind of abstraction increases the code complexity unnecessarily without providing a business (but at most a technical) benefit. Note that the MetricRegistry is already an abstraction.

My proposal would be to inject the metric registry (and probably the base metric group) into every service that shall be measured and let every service setup required structured.